libs/api/src/lib/graphql/subscription-public.ts

```diff
@@ -51,6 +52,39 @@ export const GraphQLPublicSubscription = new GraphQLObjectType<SubscriptionWithR
       resolve: createProxyingResolver(async (subscription, _, {urlAdapter}) => {
         return await urlAdapter.getSubscriptionURL(subscription)
       })
+    },
+    canExtend: {
+      type: new GraphQLNonNull(GraphQLBoolean),
+      resolve: createProxyingResolver(async (subscription, _, context) => {
+        const [paymentMethod, unpaidInvoice] = await Promise.all([
+          context.loaders.paymentMethodsByID.load(subscription.paymentMethodID),
+          context.prisma.invoice.findFirst({
+            where: {
+              subscription: {
+                userID: subscription.userID
+              },
+              OR: [{canceledAt: null}, {paidAt: null}]
+            }
+          })
+        ])
+
+        /**
+         * Can only extend when:
+         *   Subscription is extendable
+         *   Subscription is not deactivated
+         *   Subscription is about to run out (less than 1 month)
+         *   All invoices have been paid (or cancelled)
+         *   Not using a deprecated payment method
+         */
+        return (
```

Comment:
We could make this logic re-usable and refactor it in a separate function and use it among the code.

Reply:
And use it where?